tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/AIRedactionalHelper.java

```diff
@@ -42,20 +43,29 @@ public AIRedactionalHelper(ChatLanguageModel chatLanguageModel) {
         this.chatLanguageModel = chatLanguageModel;
     }
 
-    public Mono<String> suggestContent(String userInput, String mailContent) throws OpenAiHttpException, MailetException, IOException {
+    public Mono<String> suggestContent(String userInput, Optional<String> mailContent) throws OpenAiHttpException, MailetException, IOException {
         Preconditions.checkArgument(!Strings.isNullOrEmpty(userInput), "User input cannot be null or empty");
-        Preconditions.checkArgument(!Strings.isNullOrEmpty(mailContent), "Mail content cannot be null or empty");
-
-        ChatMessage systemMessage = new SystemMessage(
-                "You are an advanced email suggestion AI. Your role is to act as the recipient of an email and provide a helpful, contextually appropriate response to it. " +
-                        "Use the content of the email and incorporate the user's input to craft the response. Be polite, professional, and aligned with the tone of the email. " +
-                        "Focus on addressing the key points raised in the email, while integrating additional information or suggestions provided by the user ");
-        ChatMessage prompt = new UserMessage(
-                "Email Content: \n" + mailContent + "\n" + "User Input: \n" + userInput);
-        String llmResponse = chatLanguageModel.generate(systemMessage, prompt)
+        ChatMessage promptForContext = generatePrompt(mailContent);
+        ChatMessage promptForUserInput = new UserMessage(userInput);
+        ChatMessage promptForMailContent = new SystemMessage("[EMAIL CONTENT] (Read-only): " + mailContent);
+        String llmResponse = chatLanguageModel.generate(promptForContext, promptForUserInput, promptForMailContent)
```

Comment:
Let's use `Mono.fromCompletableFuture` + `generateAsync` to get rid of blocking calls.